metric-anomaly-data-model/src/main/proto/org/hypertrace/alert/engine/eventcondition/config/service/v1/metric_anomaly_event_condition.proto

```diff
@@ -14,6 +14,7 @@ message MetricSelection {
   MetricAggregationFunction metric_aggregation_function = 2;
   string metric_aggregation_interval = 3; //iso duration string PT15S
   Filter filter = 4;
+  string duration = 5; // iso duration string
```

Comment:
Why do we need `duration` here?

Reply:
this is the duration of metric data to be considered for rule evaluation

Reply:
So, what I was thinking is that it seems to me that the `duration` condition aligns more to  `MetricAnomalyEventCondition` than the `MetricSelection` section. The `duration` has to do more with metric data points rather than metric selection itself.

As an example, if we consider currently supported alerts.

1. alert me if avg(latency) of a service 'XYZ is greater than 10ms for the last 5 mins
2. alert me if avg(latency) of a service 'XYZ is above/below 2 bounds of historical baseline of past 1 hr data for the last 5min

```
message MetricAnomalyEventCondition {
  MetricSelection metric_selection = 1;
  repeated ViolationCondition violation_condition = 2;
  string duration = 3;
}
```

This option of putting duration with MetricSelection of combining metric selection + data points also works, but I think it is less readable and spread conditions at two places - ViloationCondition at above level, and duration condition at selection level. As an example,
- alert me if avg(latency) of a service 'XYZ for 5 mins is greater than 10ms
- alert me if avg(latency) of a service 'XYZ for 5 mins is above/below 2 bounds of historical baseline of past 1 hr data for the last 5min

Let me know, what do you think?

Reply:
Yeah, I think this makes sense, will update the schema.